docs/source/column-metadata.rst

```diff
@@ -0,0 +1,185 @@
+***************
+Column Metadata
+***************
+
+This document is heavily inspired by the KDB+ DBMS and its driver language ``q``.
+
+In ``q``, tables are constructed like this:
+
+   .. code-block:: sh
+
+      q) t: ([] name: `Alice`Bob`Joe; amount: 100 200 -400)
+
+Primary Keys
+============
+To use a primary key, we construct the table by filling in the square brackets.
+
+   .. code-block:: sh
+
+      q) t: ([id: 1 2 3] name: `Alice`Bob`Joe; amount: 100 200 -400)
+
+Foreign Keys
+============
+Many (if not all) database systems provide a way to relate columns in possibly different tables to each other. When a column in one table refers to a column in another table we call this a *foreign key*.
```

Comment:
I'll clarify this by stating at the outset that I'm only talking about relational DBMSs